node/src/tests.rs

```diff
@@ -119,3 +119,157 @@ fn purge_chain_purges_relay_and_para() {
+		.arg("--accounts")
+		.arg("3")
+		.output()
+		.unwrap();
+	let chain_spec: serde_json::Value = serde_json::from_slice(output.stdout.as_slice()).unwrap();
+	let expected = json!([
+		[
+			json!("0x3d5bd6a54d5f5292b9fb914db40cd5f7c5540f80"),
+			json!(1208925819614629200000000.0)
+		],
+		[
+			json!("0x8055e8c75a862c0da765ed0848366ef4dc492b33"),
+			json!(1208925819614629200000000.0)
+		],
+		[
+			json!("0x764d008debe9493d851d0476ccba9ec23817e2c9"),
+			json!(1208925819614629200000000.0)
+		],
+		// This is Geralds, which is also added
+		[
+			json!("0x6be02d1d3665660d22ff9624b7be0551ee1ac91b"),
+			json!(1208925819614629200000000.0)
+		]
+	]);
+
+	assert_eq!(
+		chain_spec["genesis"]["runtime"]["palletBalances"]["balances"]
+			.as_array()
+			.unwrap(),
+		expected.as_array().unwrap()
+	);
+}
+
+#[test]
+#[cfg(unix)]
+fn export_genesis_state() {
+	use nix::{
+		sys::signal::{kill, Signal::SIGINT},
+		unistd::Pid,
+	};
+
+	let base_path = tempfile::tempdir().unwrap();
+	
+	let mut cmd = Command::new(cargo_bin("moonbeam"))
+		.arg("-d")
+		.arg(base_path.path())
+		.arg("--chain")
+		.arg("local")
+		.arg("--")
+		.spawn()
+		.unwrap();
+
+	// Let it produce some blocks (roughly 20).
+	thread::sleep(Duration::from_secs(20));
+
+	let output = Command::new(cargo_bin("moonbeam"))
+		.arg("export-genesis-state")
+		.arg("--chain")
+		.arg("local")
+		.output()
+		.unwrap();
+	assert!(
+		cmd.try_wait().unwrap().is_none(),
+		"the process should still be running"
+	);
+
+	// Stop the process
+	kill(Pid::from_raw(cmd.id().try_into().unwrap()), SIGINT).unwrap();
+	assert!(wait_for(&mut cmd, 30)
+		.map(|x| x.success())
+		.unwrap_or_default());
+
+	let expected = "307830303030303030303030303030303030303030303030303030303030303030303030\
+	3030303030303030303030303030303030303030303030303030303030303030356163653533343663626566363936\
+	6231366264333431303662366363363138323137343765333035386232653933363364383336616635616234343466\
+	6235303331373061326537353937623762376533643834633035333931643133396136326231353765373837383664\
+	386330383266323964636634633131313331343030";
+	assert_eq!(expected, hex::encode(output.stdout.as_slice()))
+}
+
+#[test]
+#[cfg(unix)]
+fn export_current_state() {
+	fn run_node_and_stop() -> tempfile::TempDir {
+		use nix::{
+			sys::signal::{kill, Signal::SIGINT},
+			unistd::Pid,
+		};
+
+		let base_path = tempfile::tempdir().unwrap();
+
+		let mut cmd = Command::new(cargo_bin("moonbeam"))
+			.arg("-d")
+			.arg(base_path.path())
+			.arg("--chain")
+			.arg("local")
+			.arg("--dev-service")
+			.arg("--sealing")
+			.arg("1000")
+			.arg("--collator")
+			.arg("--author-id")
+			.arg("0x6be02d1d3665660d22ff9624b7be0551ee1ac91b")
+			.arg("--")
+			.spawn()
+			.unwrap();
+
+		// Let it produce some blocks.
+		thread::sleep(Duration::from_secs(20));
+		assert!(
+			cmd.try_wait().unwrap().is_none(),
+			"the process should still be running"
+		);
```

Comment:
For this one we do need to produce at least some blocks. But since you only check the first one, you shouldn't need to wait 20 sec (unless it takes that long to start the node?)

Also, you can change the sealing interval if you want a bunch of blocks without waiting a long time.

Reply:
In my machine I seem to need at least 5 seconds to make the test work. I will decrease it to that